exist-core/src/main/java/org/exist/xquery/functions/fn/FunHigherOrderFun.java

```diff
@@ -148,9 +148,18 @@ public Sequence eval(Sequence[] args, Sequence contextSequence)
 
             try (final FunctionReference ref = refParam) {
 				ref.analyze(cachedContextInfo);
+				if (!ref.getSignature().isOverloaded() && ref.getSignature().getArgumentCount() != 1) {
```

Comment:
If we want to bring it inline with the spec. Would it not be better just to revise our function signature for `fn:filter`?

Reply:
Do you have a pointer how? This is the current signature for the second parameter 
```java
new FunctionParameterSequenceType("function", Type.FUNCTION_REFERENCE, Cardinality.EXACTLY_ONE, "the function to call")
```

Reply:
How would one state that the provided function must have exactly one parameter of type item?

Reply:
Please note: It is the arity of the user provided filter function we are discussing here.

Reply:
Ah right of course!